File: CommonBase/src/main/java/com/common/base/callback/CallbackUI2.java
package com.common.base.callback;


import com.common.utils.UIHandler;

import androidx.annotation.NonNull;

/**
 * 2个参数，参数接口回调到UI线程
 *
 * @author LiuFeng
 * @data 2020/7/16 11:05
 */
public class CallbackUI2<T1, T2> implements CommonCallback2<T1, T2> {
    CommonCallback2<T1, T2> callback;

    public CallbackUI2(@NonNull CommonCallback2<T1, T2> callback) {
        this.callback = callback;
    }

    @Override
    public void onSuccess(T1 result1, T2 result2) {
        if (callback != null) {
            UIHandler.run(() -> callback.onSuccess(result1, result2));
        }
    }

    @Override
    public void onError(int code, String desc) {
        if (callback != null) {
            UIHandler.run(() -> callback.onError(code, desc));
        }
    }
}

File: CommonBase/src/main/java/com/common/base/callback/CallbackUI2.kt
package com.common.base.callback

import com.common.utils.thread.UIHandler

/**
 * 2个参数，参数接口回调到UI线程
 *
 * @author LiuFeng
 * @data 2020/7/16 11:05
 */
class CallbackUI2<T1, T2>(val callback: CommonCallback2<T1, T2>) : CommonCallback2<T1, T2> {

    override fun onSuccess(result1: T1, result2: T2) {
        UIHandler.run { callback.onSuccess(result1, result2) }
    }

    override fun onError(code: Int, desc: String?) {
        UIHandler.run { callback.onError(code, desc) }
    }
}